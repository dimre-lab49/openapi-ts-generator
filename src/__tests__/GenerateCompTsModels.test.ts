import { existsSync, readdirSync, readFileSync } from 'fs';
import { generateTsModels } from '../index';

beforeAll(async () => {
  await generateTsModels('https://im-wa-cmpo-nrsr.azurewebsites.net/swagger/v1/swagger.json', './output_cmp/');
});

test('Generate TS Models should generate an output directory', async done => {
  try {
    const files = await readdirSync('./output_cmp');
    expect(files.length).toBeGreaterThan(0);
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('Generate TS Models should generate an index.ts barrel file', async done => {
  try {
    const fileExist = await existsSync('./output_cmp/index.ts');
    expect(fileExist).toBeTruthy();
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('CompetencyFormGroupFac should be correct', async done => {
  try {
    const file = await readFileSync('./output_cmp/competency.form-group-fac.ts');
    expect(file.toString()).toMatchSnapshot();
    done();
  } catch (err) {
    done.fail(err);
  }
});
