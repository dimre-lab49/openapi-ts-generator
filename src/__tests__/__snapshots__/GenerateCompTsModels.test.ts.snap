// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompetencyFormGroupFac should be correct 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormGroup, Validators } from '@angular/forms';
export function CompetencyFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    isEnabled: new FormControl(''),
    name: new FormControl(''),
  });
};
"
`;
